In [1]:
## Import libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from scipy import stats # Many functions for probability distributions in python are in scipy.stats
from numpy import random
import os
import warnings
warnings.filterwarnings("ignore")

In [2]:
## load dataset
data = pd.read_csv('Survey.csv')

In [3]:
data.head()

ID  Gender  Age   Class       Major Grad Intention  GPA  Employment  \
0   1  Female   20  Junior       Other            Yes  2.9   Full-Time   
1   2    Male   23  Senior  Management            Yes  3.6   Part-Time   
2   3    Male   21  Junior       Other            Yes  2.5   Part-Time   
3   4    Male   21  Junior         CIS            Yes  2.5   Full-Time   
4   5    Male   23  Senior       Other      Undecided  2.8  Unemployed   

   Salary  Social Networking  Satisfaction  Spending Computer  Text Messages  
0    50.0                  1             3       350   Laptop            200  
1    25.0                  1             4       360   Laptop             50  
2    45.0                  2             4       600   Laptop            200  
3    40.0                  4             6       600   Laptop            250  
4    40.0                  2             4       500   Laptop            100

In [4]:
data.tail()

ID  Gender  Age      Class                   Major Grad Intention  GPA  \
57  58  Female   21     Senior  International Business             No  2.4   
58  59  Female   20     Junior                     CIS             No  2.9   
59  60  Female   20  Sophomore                     CIS             No  2.5   
60  61  Female   23     Senior              Accounting            Yes  3.5   
61  62  Female   23     Senior       Economics/Finance             No  3.2   

   Employment  Salary  Social Networking  Satisfaction  Spending Computer  \
57  Part-Time    40.0                  1             3      1000   Laptop   
58  Part-Time    40.0                  2             4       350   Laptop   
59  Part-Time    55.0                  1             4       500   Laptop   
60  Part-Time    30.0                  2             3       490   Laptop   
61  Part-Time    70.0                  2             3       250   Laptop   

    Text Messages  
57             10  
58            250  
59            500  
60             50  
61              0

In [5]:
## Summarize data set
data.describe(include ='all')

ID  Gender        Age   Class                Major  \
count   62.000000      62  62.000000      62                   62   
unique        NaN       2        NaN       3                    8   
top           NaN  Female        NaN  Senior  Retailing/Marketing   
freq          NaN      33        NaN      31                   14   
mean    31.500000     NaN  21.129032     NaN                  NaN   
std     18.041619     NaN   1.431311     NaN                  NaN   
min      1.000000     NaN  18.000000     NaN                  NaN   
25%     16.250000     NaN  20.000000     NaN                  NaN   
50%     31.500000     NaN  21.000000     NaN                  NaN   
75%     46.750000     NaN  22.000000     NaN                  NaN   
max     62.000000     NaN  26.000000     NaN                  NaN   

       Grad Intention        GPA Employment     Salary  Social Networking  \
count              62  62.000000         62  62.000000          62.000000   
unique              3        NaN          3        NaN                NaN   
top               Yes        NaN  Part-Time        NaN                NaN   
freq               28        NaN         43        NaN                NaN   
mean              NaN   3.129032        NaN  48.548387           1.516129   
std               NaN   0.377388        NaN  12.080912           0.844305   
min               NaN   2.300000        NaN  25.000000           0.000000   
25%               NaN   2.900000        NaN  40.000000           1.000000   
50%               NaN   3.150000        NaN  50.000000           1.000000   
75%               NaN   3.400000        NaN  55.000000           2.000000   
max               NaN   3.900000        NaN  80.000000           4.000000   

        Satisfaction     Spending Computer  Text Messages  
count      62.000000    62.000000       62      62.000000  
unique           NaN          NaN        3            NaN  
top              NaN          NaN   Laptop            NaN  
freq             NaN          NaN       55            NaN  
mean        3.741935   482.016129      NaN     246.209677  
std         1.213793   221.953805      NaN     214.465950  
min         1.000000   100.000000      NaN       0.000000  
25%         3.000000   312.500000      NaN     100.000000  
50%         4.000000   500.000000      NaN     200.000000  
75%         4.000000   600.000000      NaN     300.000000  
max         6.000000  1400.000000      NaN     900.000000

In [6]:
data.columns

Index(['ID', 'Gender', 'Age', 'Class', 'Major', 'Grad Intention', 'GPA',
       'Employment', 'Salary', 'Social Networking', 'Satisfaction', 'Spending',
       'Computer', 'Text Messages'],
      dtype='object')

### 2.1. For this data, construct the following contingency tables (Keep Gender as row variable)

#### 2.1.1. Gender and Major

In [7]:
data_crosstab = pd.crosstab(data['Gender'],
                            data['Major'])
data_crosstab


Major   Accounting  CIS  Economics/Finance  International Business  \
Gender                                                               
Female           3    3                  7                       4   
Male             4    1                  4                       2   

Major   Management  Other  Retailing/Marketing  Undecided  
Gender                                                     
Female           4      3                    9          0  
Male             6      4                    5          3

#### 2.1.2. Gender and Grad Intention

In [8]:
data_crosstab = pd.crosstab(data['Gender'],
                            data['Grad Intention'])
data_crosstab


Grad Intention  No  Undecided  Yes
Gender                            
Female           9         13   11
Male             3          9   17

#### 2.1.3. Gender and Employment

In [9]:
data_crosstab = pd.crosstab(data['Gender'],
                            data['Employment'])
data_crosstab


Employment  Full-Time  Part-Time  Unemployed
Gender                                      
Female              3         24           6
Male                7         19           3

#### 2.1.4. Gender and Computer 

In [10]:
data_crosstab = pd.crosstab(data['Gender'],
                            data['Computer']) 
data_crosstab


Computer  Desktop  Laptop  Tablet
Gender                           
Female          2      29       2
Male            3      26       0

## 2.2. Assume that the sample is a representative of the population of CMSU. Based on the data, answer the following question:

#### 2.2.1 What is the probability that a randomly selected CMSU student will be male?



In [11]:
data['Gender'].value_counts()

Female    33
Male      29
Name: Gender, dtype: int64

In [12]:
print('Probability that a randomly selected candidate will be male:',29/len(data['Gender']))

Probability that a randomly selected candidate will be male: 0.46774193548387094


#### 2.2.2 What is the probability that a randomly selected CMSU student will be female?


In [13]:
print('Probability that a randomly selected candidate will be female:',33/len(data['Gender']))

Probability that a randomly selected candidate will be female: 0.532258064516129


### 2.3. Assume that the sample is a representative of the population of CMSU. Based on the data, answer the following question:

#### 2.3.1 Find the conditional probability of different majors among the male students in CMSU.


In [14]:
data_crosstab = pd.crosstab(data['Gender'],
                            data['Major']) 
data_crosstab

Major   Accounting  CIS  Economics/Finance  International Business  \
Gender                                                               
Female           3    3                  7                       4   
Male             4    1                  4                       2   

Major   Management  Other  Retailing/Marketing  Undecided  
Gender                                                     
Female           4      3                    9          0  
Male             6      4                    5          3

In [15]:
print('Among MALE candidates:')
print('Probability of Accounting:',4/29)
print('Probability of CIS:',1/29)
print('Probability of Economics/Finance:',4/29)
print('Probability of International Business:',2/29)
print('Probability of Management:',6/29)
print('Probability of Other:',4/29)
print('Probability of Retailing/Marketing:',5/29)
print('Probability of Undecided:',3/29)
print()


Among MALE candidates:
Probability of Accounting: 0.13793103448275862
Probability of CIS: 0.034482758620689655
Probability of Economics/Finance: 0.13793103448275862
Probability of International Business: 0.06896551724137931
Probability of Management: 0.20689655172413793
Probability of Other: 0.13793103448275862
Probability of Retailing/Marketing: 0.1724137931034483
Probability of Undecided: 0.10344827586206896



#### P (Major | Male) = P (Major ∩ male)/ P(male)



####  2.3.2 Find the conditional probability of different majors among the female students of CMSU.


In [16]:
print('Among FEMALE candidates:')
print('Probability of Accounting:',3/33)
print('Probability of CIS:',3/33)
print('Probability of Economics/Finance	:',7/33)
print('Probability of International Business:',4/33)
print('Probability of Management:',4/33)
print('Probability of Other:',3/33)
print('Probability of Retailing/Marketing:',9/33)
print('Probability of Undecided:',0/33)
print()

Among FEMALE candidates:
Probability of Accounting: 0.09090909090909091
Probability of CIS: 0.09090909090909091
Probability of Economics/Finance	: 0.21212121212121213
Probability of International Business: 0.12121212121212122
Probability of Management: 0.12121212121212122
Probability of Other: 0.09090909090909091
Probability of Retailing/Marketing: 0.2727272727272727
Probability of Undecided: 0.0



#### P (Major | female) = P (Major ∩ female)/ P(female)



### 2.4. Assume that the sample is a representative of the population of CMSU. Based on the data, answer the following question:

#### 2.4.1 Find the probability That a randomly chosen student is a male and intends to graduate.


In [17]:
data_crosstab = pd.crosstab(data['Gender'],
                            data['Grad Intention']) 
data_crosstab


Grad Intention  No  Undecided  Yes
Gender                            
Female           9         13   11
Male             3          9   17

In [18]:
print("P(graduate ∩ Male) = P (graduate | Male) x P (male) = ",(17/29) * (29/62))


P(graduate ∩ Male) = P (graduate | Male) x P (male) =  0.27419354838709675


#### 2.4.2 Find the probability that a randomly selected student is a female and does NOT have a laptop.


In [19]:
data_crosstab = pd.crosstab(data['Gender'],
                            data['Computer']) 
data_crosstab


Computer  Desktop  Laptop  Tablet
Gender                           
Female          2      29       2
Male            3      26       0

In [20]:
print("P(laptop ∩ Female) = P (laptop | Female) x P (Female) = ",(4/33) * (33/62))


P(laptop ∩ Female) = P (laptop | Female) x P (Female) =  0.06451612903225806


### 2.5. Assume that the sample is a representative of the population of CMSU. Based on the data, answer the following question:

#### 2.5.1 Find the probability that a randomly chosen student is a male or has a full-time employment


In [21]:
data_crosstab = pd.crosstab(data['Gender'],
                            data['Employment'])
data_crosstab

Employment  Full-Time  Part-Time  Unemployed
Gender                                      
Female              3         24           6
Male                7         19           3

In [22]:
print("P(full-time ∩ Male) = P (full-time | Male) x P (male) = ",(7/29) * (29/62))


P(full-time ∩ Male) = P (full-time | Male) x P (male) =  0.11290322580645161


#### 2.5.2 Find the conditional probability that given a female student is randomly chosen, she is majoring in international business or management.
 

In [23]:
data_crosstab = pd.crosstab(data['Gender'],
                            data['Major'])
data_crosstab

Major   Accounting  CIS  Economics/Finance  International Business  \
Gender                                                               
Female           3    3                  7                       4   
Male             4    1                  4                       2   

Major   Management  Other  Retailing/Marketing  Undecided  
Gender                                                     
Female           4      3                    9          0  
Male             6      4                    5          3

In [24]:
print("P(International Business ∩ Female) = P (International Business | Female) x P (Female) = ",(4/33) * (33/62))


P(International Business ∩ Female) = P (International Business | Female) x P (Female) =  0.06451612903225806


In [25]:
print("P(Management ∩ Female) = P (Management | Female) x P (Female) = ",(4/33) * (33/62))


P(Management ∩ Female) = P (Management | Female) x P (Female) =  0.06451612903225806


In [26]:
print("P(International Business/Management ∩ Female) = P (International Business/Management | Female) x P (Female) = ",(8/33) * (33/62))


P(International Business/Management ∩ Female) = P (International Business/Management | Female) x P (Female) =  0.12903225806451613


### 2.6 Construct a contingency table of Gender and Intent to Graduate at 2 levels (Yes/No). The Undecided students are not considered now and the table is a 2x2 table. Do you think graduate intention and being female are independent events?

In [27]:
data_crosstab = pd.crosstab(data['Gender'],
                            data['Grad Intention']).drop('Undecided', axis = 1) 
data_crosstab


Grad Intention  No  Yes
Gender                 
Female           9   11
Male             3   17

For 2 events to be independent, following condition is to be satified

P(A ∩ B) = P(A) * P(B)

So, P (Grad Intention ∩ Female) = P(Grad Intention) * P(Female)

P(Female) = 33/62 = 0.532258064516129

P(Grad Intention) = 28/62 = 0.45161290322580644

P(Grad Intention) * P(Female) = 0.532258064516129 * 0.45161290322580644 = 0.24037460978147762

P (Grad Intention ∩ Female) = 11/62 = 0.1774193548387097

This is not independent events as probability multiplication of both events is not equal to combined event, so being a Graduate intention and being female candidate are not independent events.


### 2.7 Note that there are four numerical (continuous) variables in the data set, GPA, Salary, Spending and Text Messages. Answer the following questions based on the data

#### 2.7.1 If a student is chosen randomly, what is the probability that his/her GPA is less than 3?



In [28]:
data_crosstab = pd.crosstab(data['Gender'],
                            data['GPA']) 
data_crosstab


GPA     2.3  2.4  2.5  2.6  2.8  2.9  3.0  3.1  3.2  3.3  3.4  3.5  3.6  3.7  \
Gender                                                                         
Female    1    1    2    0    1    3    5    2    4    3    2    4    1    2   
Male      0    0    4    2    2    1    2    5    2    2    5    2    2    0   

GPA     3.8  3.9  
Gender            
Female    1    1  
Male      0    0

In [29]:
data[data['GPA']<3]['GPA'].count()

17

In [30]:
data[data['GPA']>=3]['GPA'].count()

45

In [31]:
#### 2.7.1 If a student is chosen randomly, what is the probability that his/her GPA is less than 3?

print('Probability that a randomly selected candidate :',17/(17+45))

Probability that a randomly selected candidate : 0.27419354838709675


#### 2.7.2 Find conditional probability that a randomly selected male earns 50 or more. Find conditional probability that a randomly selected female earns 50 or more.


In [32]:
data[data['Salary']>=50]

ID  Gender  Age      Class                   Major Grad Intention  GPA  \
0    1  Female   20     Junior                   Other            Yes  2.9   
5    6  Female   22     Senior       Economics/Finance      Undecided  2.3   
6    7  Female   21     Junior                   Other      Undecided  3.0   
7    8  Female   22     Senior                   Other      Undecided  3.1   
10  11  Female   23     Senior       Economics/Finance            Yes  2.8   
14  15    Male   21     Senior              Management            Yes  3.2   
16  17  Female   19     Junior                     CIS      Undecided  3.7   
17  18    Male   21     Junior       Economics/Finance      Undecided  3.1   
18  19    Male   19     Junior       Economics/Finance            Yes  3.5   
19  20  Female   20     Junior              Management      Undecided  3.2   
20  21  Female   22     Junior     Retailing/Marketing      Undecided  3.2   
21  22    Male   18  Sophomore              Accounting      Undecided  3.0   
22  23  Female   22     Senior     Retailing/Marketing      Undecided  3.0   
24  25  Female   20     Junior       Economics/Finance            Yes  3.0   
25  26    Male   24     Senior              Management            Yes  3.3   
26  27    Male   20     Junior       Economics/Finance            Yes  3.1   
27  28  Female   20     Junior  International Business            Yes  2.9   
28  29    Male   22     Senior     Retailing/Marketing            Yes  3.3   
30  31    Male   20     Junior              Accounting      Undecided  3.4   
35  36  Female   26     Junior              Accounting            Yes  3.3   
37  38  Female   21  Sophomore              Accounting            Yes  2.5   
38  39    Male   24     Junior       Economics/Finance            Yes  2.8   
39  40    Male   19  Sophomore     Retailing/Marketing            Yes  2.5   
40  41    Male   22     Junior              Accounting            Yes  3.2   
45  46  Female   21     Senior              Management      Undecided  3.8   
46  47  Female   20     Junior     Retailing/Marketing            Yes  3.5   
47  48    Male   19  Sophomore               Undecided      Undecided  2.5   
51  52    Male   21     Senior              Management             No  3.0   
54  55    Male   21     Senior                   Other            Yes  3.4   
55  56  Female   21     Senior     Retailing/Marketing             No  3.1   
59  60  Female   20  Sophomore                     CIS             No  2.5   
61  62  Female   23     Senior       Economics/Finance             No  3.2   

    Employment  Salary  Social Networking  Satisfaction  Spending Computer  \
0    Full-Time    50.0                  1             3       350   Laptop   
5   Unemployed    78.0                  3             2       700   Laptop   
6    Part-Time    50.0                  1             3       500   Laptop   
7    Full-Time    80.0                  1             2       200   Tablet   
10   Full-Time    50.0                  2             5       400   Laptop   
14   Part-Time    54.0                  3             4       600   Laptop   
16   Part-Time    55.0                  1             4       450   Laptop   
17   Part-Time    55.0                  2             3       600   Laptop   
18   Part-Time    52.0                  2             5       500   Laptop   
19  Unemployed    60.0                  2             6       300   Laptop   
20   Part-Time    55.0                  1             3       690   Laptop   
21  Unemployed    60.0                  1             4       600   Laptop   
22   Part-Time    55.0                  0             4       300   Laptop   
24   Part-Time    55.0                  1             3       600   Laptop   
25   Full-Time    60.0                  0             1       300   Laptop   
26   Full-Time    65.0                  1             5       375   Laptop   
27   Part-Time    50.0                  3             1       900   Laptop   
28   Part-Time    55.0   

In [33]:
data_crosstab = pd.crosstab(data['Gender'],
                            data['Salary']<50)
data_crosstab

Salary  False  True
Gender             
Female     18    15
Male       14    15

In [34]:
data_crosstab = pd.crosstab(data['Gender'],
                            data['Salary']>=50)
data_crosstab

Salary  False  True
Gender             
Female     15    18
Male       15    14

In [35]:
print('Among Male and FEMALE candidates where their earnings are more than 50 or more :')

print('Probability of male earning more than 50 or more:',14/29)
print('Probability of female earning more than 50 or more:',18/33)
print()

Among Male and FEMALE candidates where their earnings are more than 50 or more :
Probability of male earning more than 50 or more: 0.4827586206896552
Probability of female earning more than 50 or more: 0.5454545454545454



In [36]:
print("P(Salary ∩ Male) = P (Salary | Male) x P (Male) = ",(14/29) * (29/62))


P(Salary ∩ Male) = P (Salary | Male) x P (Male) =  0.22580645161290322


In [37]:
print("P(Salary ∩ Female) = P (Salary | Female) x P (Female) = ",(18/33) * (33/62))


P(Salary ∩ Female) = P (Salary | Female) x P (Female) =  0.29032258064516125


### 2.8.1 Note that there are four numerical (continuous) variables in the data set, GPA, Salary, Spending and Text Messages. For each of them comment whether they follow a normal distribution.


In [ ]:
#sns.distplot(data['GPA'])
#plt.savefig('gpa_kde.jpg');
            
sns.distplot(data['GPA']);
plt.savefig('gpa_kde.jpg');

#### Graph is not Normal Distribution

In [ ]:
sns.distplot(data['Salary']);
plt.savefig('sal_kde.jpg')

#### Graph is not Normal Distribution

In [ ]:
sns.distplot(data['Spending']);
plt.savefig('sal_kde.jpg')

#### Graph is not Normal Distribution

In [ ]:
sns.distplot(data['Text Messages']);
plt.savefig('sal_kde.jpg')

#### Graph is skewed left

#### 2.8.1 They do not follow a normal distribution for numerical (continuous) variables in the data set, GPA, Salary, Spending and Text Messages.


### 2.8.2 Write a note summarizing your conclusions


#### Probability that a randomly selected candidate will be male: 0.46774193548387094

#### The probability that a randomly chosen student is a male and intends to graduate is 0.27419354838709675

#### The probability that a randomly selected student is a female and does NOT have a laptop is 0.06451612903225806

#### The probability that a randomly chosen student is a male or has a full-time employment is 0.11290322580645161

#### The conditional probability that given a female student is randomly chosen, she is majoring in international business or management  is 0.12903225806451613

#### Being a Graduate intention and being female candidate are not independent events.

#### Probability that a randomly selected candidate that his/her GPA is less than 3  : 0.27419354838709675

#### Probability of male earning more than 50 or more: 0.4827586206896552

#### Probability of female earning more than 50 or more: 0.5454545454545454
